p4-16/spec/P4-16-spec.mdk

```diff
@@ -6834,6 +6847,19 @@ control Caller(inout Headers h) {
 }
 ~ End P4Example
 
+When a control is instantiated, all its local declarations of stateful instantiations are evaluated recursively.
+Each instantiation of a control pertains to the new instance a unique set of tables, extern objects, and inner control instances.
+Thus, invoking a control instance twice is different from invoking two control instances each once, where the former
+accesses the same stateful constructs while the latter access two different copies.
+
+This exactly-once evaluation only applies to stateful instantiations. For local variable declarations,
+whether in the `apply` block or out, and whether with initializers or not, they are always evaluated
+when a control instance is invoked. That is, local variables in a control never persist across invocations. For variables
+declared outside the `apply` block, they are evaluated at the beginning of execution.
+
+When a control with no constructor parameters is instantiated exactly once, 
+it is possible to invoke the control directly (see Section [#sec-direct-invocation]).
```

Comment:
Suggested alternate wording for the last paragraph: "The above applies to all control instantiations, regardless of whether they are explicit instantiations, or performed via direct type invocation (see Section [#sec-direct-invocation])."